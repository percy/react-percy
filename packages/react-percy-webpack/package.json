{
  "name": "@percy/react-percy-webpack",
  "version": "0.4.4",
  "repository": {
    "type": "git",
    "url": "https://github.com/percy/react-percy.git"
  },
  "license": "MIT",
  "engines": {
    "node": ">=6.0",
    "npm": ">=3.0"
  },
  "main": "lib/index.js",
  "files": [
    "lib"
  ],
  "dependencies": {
    "@percy/react-percy-test-framework": "^0.4.4",
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.2",
    "babel-plugin-react-require": "^3.0.0",
    "babel-plugin-syntax-dynamic-import": "^6.18.0",
    "babel-plugin-transform-regenerator": "^6.26.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.0",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "babel-runtime": "^6.26.0",
    "case-sensitive-paths-webpack-plugin": "^2.1.1",
    "glob": "^7.1.2",
    "memory-fs": "^0.4.1",
    "tmp": "^0.0.33",
    "webpack": "^4.0.0",
    "webpack-merge": "^3.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
