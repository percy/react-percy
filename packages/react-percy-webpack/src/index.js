export compile from './compile';
export configureEntry from './entry';
export { EntryNames, GlobalVariables, RootElementId } from './entry/constants';
export resolve from './resolve';
