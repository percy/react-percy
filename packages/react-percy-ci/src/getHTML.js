import { EntryNames, RootElementId } from '@percy/react-percy-webpack';
import findEntryPath from './findEntryPath';

const getStylesheets = assets => Object.keys(assets).filter(name => /\.css$/.test(name));

export default function getHTML(assets) {
  const stylesheets = getStylesheets(assets);

  const scripts = [
    findEntryPath(assets, EntryNames.framework),
    findEntryPath(assets, EntryNames.snapshots),
    findEntryPath(assets, EntryNames.render),
  ];

  return `
  <!DOCTYPE html>
  <html lang="en">
  <head>
      <meta charset="utf-8">
      ${stylesheets.map(stylesheet => `<link rel="stylesheet" href="${stylesheet}" />`).join('\n')}
  </head>
  <body>
    <div id="${RootElementId}"></div>
    ${scripts.map(script => `<script type="text/javascript" src="${script}"></script>`).join('\n')}
  </body>
  </html>
  `;
}
