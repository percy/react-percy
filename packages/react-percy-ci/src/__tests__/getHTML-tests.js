import { EntryNames } from '@percy/react-percy-webpack';
import getHTML from '../getHTML';

it('generates HTML with all stylesheets and the three entry scripts', () => {
  const assets = {
    '/static/styles1.css': 'styles 1 CSS',
    '/static/styles2.css': 'styles 2 CSS',
    [`/static/${EntryNames.framework}.js`]: 'framework JS',
    [`/static/${EntryNames.render}.js`]: 'render JS',
    [`/static/${EntryNames.snapshots}.js`]: 'snapshots JS',
    '/static/other.js': 'other JS',
    '/static/foo.jpg': 'foo JPG',
  };

  const html = getHTML(assets);

  expect(html).toMatchSnapshot();
});
