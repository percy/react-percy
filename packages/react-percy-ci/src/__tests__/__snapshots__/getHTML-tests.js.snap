// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates HTML with all stylesheets and the three entry scripts 1`] = `
"
  <!DOCTYPE html>
  <html lang=\\"en\\">
  <head>
      <meta charset=\\"utf-8\\">
      <link rel=\\"stylesheet\\" href=\\"/static/styles1.css\\" />
<link rel=\\"stylesheet\\" href=\\"/static/styles2.css\\" />
  </head>
  <body>
    <div id=\\"percy-root\\"></div>
    <script type=\\"text/javascript\\" src=\\"/static/percy_framework.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/static/percy_snapshots.js\\"></script>
<script type=\\"text/javascript\\" src=\\"/static/percy_render.js\\"></script>
  </body>
  </html>
  "
`;
